### PR TITLE
feat(config): Rename packageRules selectors

### LINKED ISSUE #6805: Rename packageRule selectors
Many users make the `packageRules` mistake of either (a) no matching rules or (b) nothing "to do" once they've matched. What if we renamed all the selectors/matchers to start with `match`?

Before:

```js
  let {
    paths,
    languages,
    baseBranchList,
    managers,
    datasources,
    depTypeList,
    packageNames,
    packagePatterns,
    excludePackageNames,
    excludePackagePatterns,
    matchCurrentVersion,
    sourceUrlPrefixes,
    updateTypes,
  } = packageRule;
```

After:

```js
  let {
    matchPaths,
    matchLanguages,
    matchBaseBranches,
    matchManagers,
    matchDatasources,
    matchDepTypes,
    matchPackageNames,
    matchPackagePatterns,
    excludePackageNames,
    excludePackagePatterns,
    matchCurrentVersion,
    matchSourceUrlPrefixes,
    matchUpdateTypes,
  } = packageRule;
```

Then we could more easily explain "Each package rule must have at least one match|exclude option and at least one non match|exclude option" and also more easily "lint" them and raise warnings.

### PR DESCRIPTION
**Pull request by zharinov**
Closes #6805

### CODE REVIEW

File: lib/config/common.ts
Comment (zharinov) [inline lib/config/common.ts:37]: Haven't found its usages after introducing `matchManagers` in `PackageRule` interface

File: lib/config/common.ts
Comment (viceice) [inline lib/config/common.ts:211]: Shouldn't this be `matchSourceUrlsPrefixes`? Don't we do prefix match as intended?

File: lib/config/massage.ts
Comment (viceice) [inline lib/config/massage.ts:62]: When TS 4.0 is available we can reduce it to:
```suggestion
          (newRule.matchUpdateTypes ??= []).push(key);
```
 🙃 

https://devblogs.microsoft.com/typescript/announcing-typescript-4-0-beta/#short-circuiting-assignment-operators

File: lib/config/presets/index.ts
Comment (viceice) [inline lib/config/presets/index.ts:170]: Looks a little ugly to me 🤔

File: renovate.json
Comment (viceice) [inline renovate.json:31]: We should defer this until the app is updated to new version

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/config/massage.ts
Comment (zharinov) [reply lib/config/massage.ts:62]: Exactly what I thought reading release notes :)

File: lib/config/common.ts
Comment (rarkins) [reply lib/config/common.ts:211]: We do, but I was thinking it was too long of a name. I guess prefixes is better explanatory though. @zharinov can you make it `matchSourceUrlPrefixes` instead?

File: lib/config/presets/index.ts
Comment (rarkins) [reply lib/config/presets/index.ts:170]: So the alternative is `exclude` which makes other logic less clear. e.g. docs need to say "you must have at least one `matchX` or `excludeX`, etc. Either way introduces readability challenges, and I don't mind either way.

File: renovate.json
Comment (rarkins) [reply renovate.json:31]: I updated it today

Comment (viceice) [reply renovate.json:31]: But we need to defer this change until this pr is gone life in the app, otherwise we don't get any new updates because of a wrong renovate config

File: lib/config/presets/index.ts
Comment (viceice) [reply lib/config/presets/index.ts:170]: ok

File: renovate.json
Comment (rarkins) [reply renovate.json:31]: Oh, sorry - I see what you mean. Yes, let's revert this from this PR

**Review by rarkins — CHANGES_REQUESTED** [review]
Revert changes to renovate.json

**Review by rarkins — COMMENTED** [review]
LGTM, but let's hold off a little until I am ready to merge it into the app and pro so we don't end up confusing people with documentation

File: lib/config/common.ts
Comment (viceice) [inline lib/config/common.ts:209]: Maybe we can remove those `exclude**` options and simply use a negation pattern, like prefix with `!` ?

Comment (rarkins) [reply lib/config/common.ts:209]: Should be a separate PR then, probably first. Negation should suit package names ok, not sure about patterns though as presumably it's allowed to start a regex with `!`

**Review by viceice — DISMISSED** [review]

**Comment by zharinov** [comment]
- [ ] Do we still need it?
- [ ] Resolve conflicts
- [ ] Rename new options (if any)

**Comment by viceice** [comment]
> * [ ]  Do we still need it?

yes, should make config options more clear

**Comment by rarkins** [comment]
@zharinov can you check this over again? I have tried to resolve the conflicts without breaking anything

**Comment by zharinov** [comment]
This one should be fine now, I guess

File: lib/workers/repository/process/lookup/__snapshots__/index.spec.ts.snap
Comment (viceice) [inline lib/workers/repository/process/lookup/__snapshots__/index.spec.ts.snap:219]: Why this change? 🤔

Comment (zharinov) [reply lib/workers/repository/process/lookup/__snapshots__/index.spec.ts.snap]: Fixed

**Review by viceice — DISMISSED** [review]

**Comment by Diaoul** [comment]
New to renovate, I just configured using those awesome new names following the docs but it seems it is not yet deployed to the bots. What's the ETA on this? Is there a way to specify the version in `renovate.json`?
```
File: .github/renovate.json5
Error type: The renovate configuration file contains some invalid settings
Message: Invalid configuration option: packageRules[0].matchDatasources, Invalid configuration option: packageRules[1].matchDatasources, Invalid configuration option: packageRules[2].matchDatasources, Invalid 
```
Thanks!

**Comment by rarkins** [comment]
@Diaoul sorry! The app usually adopts new releases pretty quickly but I was thinking this was a major rename and I might skip deploying late on a Friday. I hadn't thought that the docs would update automatically and it's pretty amazing that you managed to find this PR and work out what went wrong.

**Comment by rarkins** [comment]
I'm putting it live now but there's some possibility I may need to roll back if I noticed any undesirable behavior due to the naming migration

**Comment by Diaoul** [comment]
I can confirm it is now working. Thanks! :tada:

**Comment by roock** [comment]
@rarkins shouldn't have been this a major version update as this breaks existing configurations?

**Comment by viceice** [comment]
@roock We have migrations for this, so this shouldn't break. Please open a new discussion if you have any problems.

**Pull request merged**
